packages/effect/src/GlobalValue.ts

```diff
@@ -46,6 +42,12 @@ const globalStore = (globalThis as any)[globalStoreId] as Map<unknown, any>
  * @since 2.0.0
  */
 export const globalValue = <A>(id: unknown, compute: () => A): A => {
+  if (!globalStore) {
+    // @ts-expect-error
+    globalThis[globalStoreId] = new Map()
```

Comment:
```suggestion
    globalThis[globalStoreId] ??= new Map()
```

Not sure if we want to overwrite if it already exists.

Reply:
how can it already exist if `globalStore` is undefined?

Reply:
It would in the case of a duplicated module.